opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -103,11 +107,40 @@ namespace Opm {
             }
         }
 
+        // update the previous well state. This is used to restart failed steps.
+        previous_well_state_ = well_state_;
+
+
+        if (wellCollection().havingVREPGroups() ) {
+            rateConverter_->template defineState<ElementContext>(ebosSimulator_);
```

Comment:
Will the `reset` below make `defineState` not helpful anymore?

Reply:
You are right. These lines should be called pr timestep

Reply:
Actually, these lines can be removed since its already called in prepareTimeStep()